File: plugins/claude-md-plugin/core/tests/fixtures/java/SealedState.java
package com.example.state;

public sealed class State permits Idle, Loading, Loaded, Error {
}

final class Idle extends State {
}

final class Loading extends State {
    private final int progress;
    public Loading(int progress) { this.progress = progress; }
}

final class Loaded extends State {
    private final String data;
    public Loaded(String data) { this.data = data; }
}

final class Error extends State {
    private final String message;
    public Error(String message) { this.message = message; }
}

File: plugins/claude-md-plugin/core/tests/fixtures/kotlin/SealedState.kt
package com.example.state

sealed class State {
    object Idle : State()
    data class Loading(val progress: Int) : State()
    data class Loaded(val data: String) : State()
    data class Error(val message: String) : State()
}

sealed interface Event {
    object Start : Event
    data class DataReceived(val data: String) : Event
    object Cancel : Event
}
